src/engine/enginebuffer.cpp

```diff
@@ -1136,7 +1151,7 @@ void EngineBuffer::processSlip(int iBufferSize) {
     bool enabled = static_cast<bool>(load_atomic(m_slipEnabled));
     if (enabled != m_bSlipEnabledProcessing) {
         m_bSlipEnabledProcessing = enabled;
-        if (enabled) {
+        if (enabled || m_slipCancelled.fetchAndStoreAcquire(0)) {
```

Comment:
I think it's possible, although unlikely, that the following could happen, especially at high latencies:

1. processslip is called, completes
2. user pushes slip enable
3. user pushes slip cancel
4. user disables slip
5. process slip is called again.

In this situation, slip is disabled, but cancel is true.  The code to check slipcancel is never called, and the slip cancel state is stuck on.  The next time the user tries to turn on slip, it will immediately cancel.  Therefore I think every time slip is disabled, slipcancel should also be set to zero.